### PR TITLE
Normalize device objects when passed into constructors

### PR DESCRIPTION
**Pull request by Chillee**
Fixes https://github.com/pytorch/torchdynamo/issues/633

### CODE REVIEW

File: torchinductor/scheduler.py [hunk 1]
```diff
@@ -554,6 +554,8 @@ def draw_buffers(nodes, fname, print_graph=False):
         if "fusion_meta" not in node.meta:
             continue
         group = node.meta["fusion_meta"].group
+        if isinstance(group, tuple):
+            group = group[1]
```
Comment (ngimel) [inline torchinductor/scheduler.py:558]: what's `group[1]`?

**Review by ngimel — APPROVED** [review]

File: torchinductor/scheduler.py (hunk 1 above)
Comment (Chillee) [reply torchinductor/scheduler.py:558]: It's the "non-device" part of group. Group looks like (device, indexing_info).

File: torchinductor/lowering.py [hunk 2]
```diff
@@ -111,6 +111,8 @@ def decode_device(device):
         return torch.tensor(0.0).device  # default device
     if isinstance(device, str):
         return torch.device(device)
+    if device.type == "cuda" and device.index is None:
+        return torch.device("cuda", index=torch.cuda.current_device())
```
Comment (eellison) [inline torchinductor/lowering.py:115]: do we guard on current device anywhere ?

File: torchinductor/lowering.py [hunk 3]
```diff
@@ -111,6 +111,8 @@ def decode_device(device):
         return torch.tensor(0.0).device  # default device
     if isinstance(device, str):
         return torch.device(device)
```
Comment (jansel) [inline torchinductor/lowering.py:113]: We should normalize this one too:
```
>>> torch.device("cuda")
device(type='cuda')
```

**Review by jansel — CHANGES_REQUESTED** [review]

File: torchinductor/lowering.py (hunk 3 above)
Comment (Chillee) [reply torchinductor/lowering.py]: done

File: torchinductor/lowering.py (hunk 2 above)
Comment (jansel) [reply torchinductor/lowering.py:115]: We don't, though seems fine to do that in another PR.

**Review by jansel — APPROVED** [review]
